pkg/client/unversioned/clientcmd/config.go

```diff
@@ -75,8 +77,16 @@ func (o *PathOptions) GetLoadingPrecedence() []string {
 	if envVarFiles := o.GetEnvVarFiles(); len(envVarFiles) > 0 {
 		return envVarFiles
 	}
-
-	return []string{o.GlobalFile}
+	files, err := loadFileOrDir(o.GlobalFile)
+	// If GlobalFile doesn't exist, try GlobalFileDir
```

Comment:
See the comment from @smarterclayton here:

https://github.com/kubernetes/kubernetes/pull/38861#issuecomment-271637115